src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV1.cs

```diff
@@ -47,6 +48,68 @@ protected static class ArtifactEventProperties
         }
     }
 
+        // Caller: PublishPipelineArtifact task
+    // Can be invoked from a build run or a release run should a build be set as the artifact. 
+    public class PublishPipelineArtifactTaskV1 : PipelineArtifactTaskPluginBaseV1
+    {
+        public override Guid Id => PipelineArtifactPluginConstants.PublishPipelineArtifactTaskId;
+
+        public static readonly Regex jobIdentifierRgx = new Regex("[^a-zA-Z0-9 - .]", RegexOptions.Compiled | RegexOptions.CultureInvariant);
+
+        protected override async Task ProcessCommandInternalAsync(
+            AgentTaskPluginExecutionContext context, 
+            CancellationToken token)
+        {           
+            string artifactName = context.GetInput(ArtifactEventProperties.ArtifactName, required: false);
+            string targetPath = context.GetInput(TargetPath, required: true);
+            string hostType = context.Variables.GetValueOrDefault("system.hosttype")?.Value; 
+            if (!string.Equals(hostType, "Build", StringComparison.OrdinalIgnoreCase)) {
+                throw new InvalidOperationException(
+                    StringUtil.Loc("CannotUploadFromCurrentEnvironment", hostType ?? string.Empty)); 
+            }
+
+            if (String.IsNullOrWhiteSpace(artifactName))
+            {
+                string jobIdentifier = context.Variables.GetValueOrDefault("system.jobIdentifier").Value;
+                var normalizedJobIdentifier = NormalizeJobIdentifier(jobIdentifier);
+                artifactName = normalizedJobIdentifier;
+            }
+
+            // Project ID
+            Guid projectId = new Guid(context.Variables.GetValueOrDefault(BuildVariables.TeamProjectId)?.Value ?? Guid.Empty.ToString());
+            ArgUtil.NotEmpty(projectId, nameof(projectId));
+
+            // Build ID
+            string buildIdStr = context.Variables.GetValueOrDefault(BuildVariables.BuildId)?.Value ?? string.Empty;
+            if (!int.TryParse(buildIdStr, out int buildId))
+            {
+                // This should not happen since the build id comes from build environment. But a user may override that so we must be careful.
+                throw new ArgumentException(StringUtil.Loc("BuildIdIsNotValid", buildIdStr));
+            }
+
+            string fullPath = Path.GetFullPath(targetPath);
```

Comment:
Assuming `$(System.DefaultWorkingDirectory)` is `c:\agent\_work\1\s`, here are a few examples of how this should work ...

| `path` value | Resolved path |
|---|----|
| `.` | `c:\agent\_work\1\s` |
| `.\a.txt` | `c:\agent\_work\1\s\a.txt` |
| `a.txt` | `c:\agent\_work\1\s\a.txt` |
| `..\b.dll` | `c:\agent\_work\1\b.dll` |
| `$(Pipeline.Workspace)/c` | `c:\agent\_work\1\c` |
| `d:\temp\a` | `d:\temp\a` |

Reply:
@jahsu-MSFT - think this will end up looking something like this ... 

```cs
    string fullPath = Path.IsPathFullyQualified(targetPath) ?
        targetPath :
        Path.GetFullPath(
            targetPath,
            context.Variables.GetValueOrDefault(Constants.Variables.System.DefaultWorkingDirectory));

    if (!Directory.Exists(fullPath) && !File.Exists(fullPath))
    {
        // if local path is neither file nor folder
        throw new FileNotFoundException(StringUtil.Loc("PathNotExist", targetPath));
    }   
```

Reply:
Interesting.. looks like the default working directory is not inside of the _work folder
![image](https://user-images.githubusercontent.com/45610531/58820141-a775d380-85e6-11e9-81c6-8d2c58b36e87.png)

Reply:
Is it possible you named your agent's work directory `y` instead of accepting the default of `_work`?

(this is one of the questions prompted by the config command)

Reply:
Yes, that's correct. work directory name is y there for _work is y in my system

Reply:
Nevermind. I thought by default we're creating a _work folder as the working directory and then realized it's using the default work folder name